### PR TITLE
[HOST][MODULAR]Adds a pointing cooldown (once per half second)

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

Adds a cooldown to pointing. Three points per 10 seconds. No more point spamming. 

https://user-images.githubusercontent.com/77420409/179305509-ac40c11e-0d8c-443d-b65a-9767da34aa0b.mp4

## How This Contributes To The Skyrat Roleplay Experience

Less point spamming, yeah. 

## Changelog

:cl:
code: Overrides pointing code to add a timer on pointing. 
/:cl:

### CODE REVIEW

File: code/modules/client/verbs/pointing_cooldown.dm [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	if(pointstacks)
+		pointstacks--
+	addtimer(VARSET_CALLBACK(src, pointstacks, 3), 10 SECONDS, TIMER_OVERRIDE)
+
+/mob/pointed()
+	if(client.pointstacks)
+		. = ..()
+		SEND_SIGNAL(client, COMSIG_MOB_POINTED)
+	else
+		to_chat(usr, "You are pointing too hard!")
```
Comment (tf-4) [inline code/modules/client/verbs/pointing_cooldown.dm]:
```suggestion
		balloon_alert(usr, "can't point yet!")
```

Comment (ghost) [reply code/modules/client/verbs/pointing_cooldown.dm]: Alternatively something like 'slow down!' would work

File: code/modules/client/verbs/pointing_cooldown.dm [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/client
+	var/pointstacks = 3
```
Comment (Zonespace27) [inline code/modules/client/verbs/pointing_cooldown.dm]: code doc comment

File: code/modules/client/verbs/pointing_cooldown.dm [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	RegisterSignal(src, COMSIG_MOB_POINTED, /client.proc/point_cooldown)
+
+/client/proc/point_cooldown()
+	SIGNAL_HANDLER
+	if(pointstacks)
+		pointstacks--
+	addtimer(VARSET_CALLBACK(src, pointstacks, 3), 10 SECONDS, TIMER_OVERRIDE)
+
+/mob/pointed()
+	if(client.pointstacks)
```
Comment (Zonespace27) [inline code/modules/client/verbs/pointing_cooldown.dm]:
```suggestion
	if(client?.pointstacks)
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/client/verbs/pointing_cooldown.dm (hunk 2 above)
Comment (tf-4) [inline code/modules/client/verbs/pointing_cooldown.dm]:
```suggestion
	var/point_stacks = 3
```

**Comment by GuiltyNeko** [comment]
~~I'm just a guy with an opinion but unironically balloon alerts would be best for this.~~

Already suggested, muh bad.

File: code/modules/client/verbs/pointing_cooldown.dm [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/client
+	/// Your clients allowed pointing stacks.
+	var/point_stacks = 3
```
Comment (jjpark-kb) [inline code/modules/client/verbs/pointing_cooldown.dm]: make 3 a define

File: code/modules/client/verbs/pointing_cooldown.dm [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+	. = ..()
+	RegisterSignal(src, COMSIG_MOB_POINTED, /client.proc/point_cooldown)
+/**
+ * Increments the allowed pointing stacks and resets the pointer cooldown.
+ */
+/client/proc/point_cooldown()
+	SIGNAL_HANDLER
+	if(point_stacks)
+		point_stacks--
+	addtimer(VARSET_CALLBACK(src, point_stacks, 3), 10 SECONDS, TIMER_UNIQUE)
```
Comment (jjpark-kb) [inline code/modules/client/verbs/pointing_cooldown.dm]: make 3 a define

**Comment by ArdentSpark** [comment]
three per ten is real harsh, not being to point for an entire ten seconds, seems incredibly high. Bearing in mind you might be in combat.
Something like one per two isn't that much different, or hell one per three would stop you getting locked out for so long

File: code/modules/client/verbs/pointing_cooldown.dm [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+/mob/pointed()
+	if(client?.point_stacks)
+		. = ..()
+		SEND_SIGNAL(client, COMSIG_MOB_POINTED)
+	else
+		balloon_alert(usr, "too fast, calm down!")
```
Comment (tf-4) [inline code/modules/client/verbs/pointing_cooldown.dm]: This is kind of long and not necessarily clear, I still prefer

```suggestion
		balloon_alert(usr, "can't point yet!")
```

**Review by tf-4 — APPROVED** [review]

File: code/modules/client/verbs/pointing_cooldown.dm [hunk 7]
```diff
@@ -0,0 +1,26 @@
+#define POINT_STACK_LIMIT 3
+/client
+	/// Your clients allowed pointing stacks.
+	var/point_stacks = POINT_STACK_LIMIT
```
Comment (Gandalf2k15) [inline code/modules/client/verbs/pointing_cooldown.dm]: making this var on every client is going to add pointless overhead.

Comment (nevimer) [reply code/modules/client/verbs/pointing_cooldown.dm]: what do otherwise?

Comment (YakumoChen) [reply code/modules/client/verbs/pointing_cooldown.dm]: > making this var on every client is going to add pointless overhead.

> pointless

ba dum tish

**Comment by YakumoChen** [comment]
3 per 10 is way too strange of a cooldown. Just make it one per 3 or something

**Comment by nevimer** [comment]
> 3 per 10 is way too strange of a cooldown. Just make it one per 3 or something

why would I do that?

File: code/modules/client/verbs/pointing_cooldown.dm (hunk 7 above)
Comment (nevimer) [reply code/modules/client/verbs/pointing_cooldown.dm]: > making this var on every client is going to add pointless overhead.

ok but how else are you going to track something on a per client basis

**Comment by YakumoChen** [comment]
having to wait a full 10 seconds after pointing a few times is much more obnoxious. Being able to point once every three seconds feels a lot more sensible and is more in line with other similar systems such as emoting.

**Review by tf-4 — CHANGES_REQUESTED** [review]
one per three does make more sense IMO.

**Comment by nevimer** [comment]
> one per three does make more sense IMO.

That is less and that is not the request that was made.

**Comment by YakumoChen** [comment]
1 per 3.3 seconds then?

I don't believe @hugmeorelse will mind as long as the intent of the request (reducing spam pointing) is honoured, and the cooldown is generally the same.

Potentially having to wait 10 seconds to point again is a social ice age and far more likely to stifle RP, being able to spam point 3 times in a second does not make it any more noticable and is more annoying.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
After discussing this with Xyel, I'd like to try this out with one point every 0.5 seconds, so as not to heavily punish those that use it for legitimate reasons, while preventing spam in a relatively decent way, making it even more obvious when someone's just spamming it to be annoying.

**Comment by nevimer** [comment]
oh woops forgot about this lol

**Comment by tf-4** [comment]
> oh woops forgot about this lol

I can reopen it if you're ready to make the requested changes

**Comment by nevimer** [comment]
> > oh woops forgot about this lol
> 
> I can reopen it if you're ready to make the requested changes

ya sure ill do it

**Review by tf-4 — CHANGES_REQUESTED** [review]
Needs to be in a modular folder

**Comment by nevimer** [comment]
https://user-images.githubusercontent.com/77420409/190879737-b0df80da-5782-4f86-863f-3934182c9264.mp4

please enjoy

**Review by tf-4 — APPROVED** [review]

**Pull request merged**